src/platform_impl/android/mod.rs

```diff
@@ -1012,46 +1003,49 @@ impl MonitorHandle {
         self.app.config().density().map(|dpi| dpi as f64 / 160.0).unwrap_or(1.0)
     }
 
-    pub fn refresh_rate_millihertz(&self) -> Option<u32> {
-        // FIXME no way to get real refresh rate for now.
-        None
-    }
-
-    pub fn video_modes(&self) -> impl Iterator<Item = VideoModeHandle> {
-        let size = self.size().into();
-        // FIXME this is not the real refresh rate
-        // (it is guaranteed to support 32 bit color though)
-        std::iter::once(VideoModeHandle {
-            size,
+    pub fn current_video_mode(&self) -> Option<VideoModeHandle> {
+        Some(VideoModeHandle {
+            size: screen_size(&self.app),
+            // FIXME: it is guaranteed to support 32 bit color though
             bit_depth: 32,
-            refresh_rate_millihertz: 60000,
             monitor: self.clone(),
         })
     }
+
+    pub fn video_modes(&self) -> impl Iterator<Item = VideoModeHandle> {
+        self.current_video_mode().into_iter()
+    }
 }
 
 #[derive(Clone, Debug, Eq, Hash, PartialEq)]
 pub struct VideoModeHandle {
-    size: (u32, u32),
+    size: PhysicalSize<u32>,
     bit_depth: u16,
-    refresh_rate_millihertz: u32,
     monitor: MonitorHandle,
 }
 
 impl VideoModeHandle {
     pub fn size(&self) -> PhysicalSize<u32> {
-        self.size.into()
+        self.size
     }
 
     pub fn bit_depth(&self) -> u16 {
         self.bit_depth
     }
 
-    pub fn refresh_rate_millihertz(&self) -> u32 {
-        self.refresh_rate_millihertz
+    pub fn refresh_rate_millihertz(&self) -> Option<NonZeroU32> {
+        None
     }
 
     pub fn monitor(&self) -> MonitorHandle {
         self.monitor.clone()
     }
 }
+
+fn screen_size(app: &AndroidApp) -> PhysicalSize<u32> {
+    if let Some(native_window) = app.native_window() {
```

Comment:
Perhaps we could pull this from the `configuration` instead, but it's all DPI-dependent:

https://docs.rs/ndk/latest/ndk/configuration/struct.Configuration.html#method.screen_width_dp

Reply:
Let me know if you want me to add any comment for that.

Reply:
I think I/we should just play with that idea, but maybe for now let the size of the mode always return `None` or something, to disconnect it from a possible surface?

Reply:
AFAIK `VideoModeHandle::size()` shouldn't be used to create surfaces anyway, they should just tell you the the resolution in physical pixels of that video mode.

I will add that to the documentation.
Is that enough or is the information exposed here just wrong?

Reply:
Exactly that as well. The surface will barely ever be the fill resolution of the screen when statusbars and navbars are removed from the screen.

Reply:
I've changed `VideoModeHandle::size()` to return an `Option` now.